python/desc/twinkles/validate_ic.py

```diff
@@ -0,0 +1,849 @@
+
+        print('------------')
+
+        return
+
+    def compare_sne_lens_mags(self, spr_sne_lens_df):
+
+        """
+        This test compares the things that get swapped in directly from the input catalog.
+        """
+
+        df = pd.read_csv(os.path.join(os.environ['TWINKLES_DIR'], 'data',
+                                      'dc2_sne_cat.csv'))
+
+        lens_mag_error = []
+
+        for lens_gal_row in spr_sne_lens_df.iterrows():
+            lens_idx, lens_gal_df = lens_gal_row
+            u_id = lens_gal_df['uniqueId']
+
+            lens = df.query('twinkles_sysno == %i' % lens_gal_df['twinkles_system'])
+
+            # Even though SNe images don't appear in every image the lens galaxy should always
+            # be in the instance catalog
+            
+            lens_mag_error.append(lens['bulge_magnorm'].iloc[0] -
+                                  lens_gal_df['phosimMagNorm'])
+
+        max_lens_mag_error = np.max(np.abs(lens_mag_error))
+
+        print('------------')
+        print('SNe Lens Magnitude Test Results:')
+
+        if max_lens_mag_error < 0.01:
+            print('Pass: Max lens phosim MagNorm error less than 0.01 mags.')
+        else:
+            raise CatalogError('\nFail: Max lens phosim MagNorm error is greater than 0.01 mags. ' + 
+                               'Max error is: %.4f mags.' % max_lens_mag_error)
+
+        print('------------')
+
+        return
+
+    def compare_sne_image_inputs(self, spr_sne_df, spr_sne_lens_df, visit_mjd,
+                                 sne_file_loc):
+
+        """
+        This test compares the image inputs from catalog.
+        """
+
+        df = pd.read_csv(os.path.join(os.environ['TWINKLES_DIR'], 'data',
+                                      'dc2_sne_cat.csv'))
+
+        errors_present = False
+        z_s_error = False
+        sed_error = False
+        errors_string = 'Errors in:'
+
+        for lens_gal_row in spr_sne_lens_df.iterrows():
+
+            lens_idx, lens_gal_df = lens_gal_row
+            u_id = lens_gal_df['uniqueId']
+
+            spr_sys_df = spr_sne_df.query('lens_galaxy_uID == %i' % u_id)
+
+            # There may be no images present in the current instance catalog
+            if len(spr_sys_df) == 0:
+                continue
+
+            lens = df.query('twinkles_sysno == %i' % spr_sys_df['twinkles_system'].iloc[0])
+            img_vals = spr_sys_df['image_number'].values
+
+            for idx, image_on in list(enumerate(img_vals)):
+                
+                if lens['zs'].iloc[image_on] - spr_sys_df['redshift'].iloc[idx] > 0.005:
+                    if z_s_error is False:
+                        errors_string = errors_string + "\nSNe Image Redshifts. First error found in gal_id: %i " % u_id
+                        errors_present = True
+                        z_lens_error = True
+
+                sed_name = '%s/specFileGLSN_%i_%i_%.4f.txt.gz' % (sne_file_loc,
```

Comment:
We should also check that the SNe SED files exist, since there is a risk that they won't get written out properly.